### DISCUSSION TITLE
Rooting UART method - multiple login prompts

### DISCUSSION THREAD

**Discussion by hannesdonel** [Hypfer/Valetudo#2063] [Q&A / Support] [answered]
### Are you already using Valetudo?

Yes, for a long time

### Your questions

My old Roborock S5 (with Valetudo) gets replaced. While trying to root my new Xiaomi X10+ (bought April 2024 via Xiaomi Store), I run into following behavior:

**Reproduction:**
1. Wire up robot with PBC breakout an start logging with `sudo screen /dev/ttyUSB0 115200, ixoff`
2. Power on robot, logging is visible, `root password changed` log appears
3. Wait until logging stops and insert USB stick, apply OTG jumper
4. Get presented to **three** (it's always three!) lines saying `p2114_release login`
5. Type `root`, hit enter
6. Now it sometimes asks for my password, sometimes again presents to `p2029_release login` and then freezes
7. If asked for password type password, hit enter
8. Then two things can happen: It asks again for password and then freezes or says `Login incorrect` and freezes
9. No matter what happened before it prompts two times `Login Timeout after 60 seconds`

**Password creation**
1. Serial Number is confirmed from Xiaomi and has following format: `41717/BF*******08582`
2. Password is created with `echo -n "41717/BF*******08582" | md5sum | base64`

**USB flashing**
1. `fdisk /dev/sdX` and format it like this:

- Type `o` to create a new empty DOS partition table
- Type `n` to create a new partition
- Choose primary partition by typing `p`
- Use the entire disk for partition
- Type `w` to write changes.

2. `mkfs.fat -F 32 -S 4096 /dev/sdX`
3. `dd if=path/to/image/dreame_uart_root.img of=/dev/sdX && sync`
4. `dreame_uart_root.img` provided in Valetudo docs is used
5. I tried two different sticks

I'm thankful for every experience, hint and help what to do.

**Comment by vivia** [comment, marked as answer]
Check the two following things:

1. Make sure your USB drive has only one partition. In order to be sure, delete all partitions, leave the USB drive completely blank, and then flash the `dreame_uart_root.img` from the Valetudo docs.
2. Double-check the solderings on the PCB side, especially those related to the USB drive. Make sure that it doesn't disconnect and reconnect.

**Reply by hannesdonel** [reply]
Thank you very much. Your comment actually send me back to carefully review my USB stick. Solution was do explicitly format it as MBR before flashing it.

**Reply by csutakbalazs** [reply]
@hannesdonel I have been struggling for hours with the same issue (except there were always two prompts for me), even though I triple-checked, that the pendrive was formatted as MBR. Following your advice, I deleted all partitions, explicitly reformatted it as MBR, reflashed the image, and everything worked seamlessly afterward. Thank you so much for returning and leaving this comment here!
